Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.py

```diff
@@ -0,0 +1,658 @@
+    markdown = tableToMarkdown('Asset Internet Exposure', parsed, removeNull=True,
+                               headerTransform=string_to_table_header)
+    command_results = CommandResults(
+        outputs_prefix='ASM.AssetInternetExposure',
+        outputs_key_field='asm_ids',
+        outputs=parsed,
+        raw_response=response,
+        readable_output=markdown
+    )
+
+    return command_results
+
+
+def fetch_incidents(client: Client, max_fetch: int, last_run: Dict[str, int],
+                    first_fetch_time: Optional[int], severity: Optional[list]
+                    ) -> Tuple[Dict[str, int], List[dict]]:
+    """
+    This function will execute each interval (default is 1 minute).
+
+    Args:
+        client (Client): CortexXpanse client to use.
+        max_fetch (int): Maximum numbers of incidents per fetch.
+        last_run: The greatest incident created_time we fetched from last fetch
+        first_fetch_time: If last_run is None then fetch all incidents since first_fetch_time
+        severity: The severity of the alerts that will be fetched.
+
+    Returns:
+        next_run: This will be last_run in the next fetch-incidents
+        incidents: Incidents that will be created in Cortex XSOAR
+    """
+    # Get the last fetch time, if exists
+    last_fetch = last_run.get('last_fetch', None)
+
+    # Handle first time fetch
+    if last_fetch is None:
+        last_fetch = first_fetch_time
+    else:
+        last_fetch = int(last_fetch)
+
+    latest_created_time = cast(int, last_fetch)
+    incidents = []
+
+    # Changed from 'last_fetch' to 'latest_created time' because they are the same and fixed type error.
+    filters = [{'field': 'alert_source', 'operator': 'in', 'value': ['ASM']}, {
+        'field': 'creation_time', 'operator': 'gte', 'value': latest_created_time + 1}]
+    if severity:
+        filters.append({"field": "severity", "operator": "in", "value": severity})
+
+    request_data = {'request_data': {'filters': filters, 'search_from': 0,
+                                     'search_to': max_fetch, 'sort': {'field': 'creation_time', 'keyword': 'asc'}}}
+
+    raw = client.list_alerts_request(request_data)
+
+    items = raw.get('reply', {}).get('alerts')
+    for item in items:
+        # for item in items.outputs:
+        incident_created_time = item['detection_timestamp']
+        incident = {
+            'name': item['name'],
+            'details': item['description'],
+            'occurred': timestamp_to_datestring(incident_created_time),
+            'rawJSON': json.dumps(item),
+            'severity': convert_to_demisto_severity(item.get('severity', 'Low'))
+        }
+
+        incidents.append(incident)
+
+        if incident_created_time > latest_created_time:
+            latest_created_time = incident_created_time
+
+    next_run = {'last_fetch': latest_created_time}
+    return next_run, incidents
+
+
+def test_module(client: Client) -> None:
```

Comment:
in this method check if the integration fetching incidents(`if params.get('isFetch', False):`)
and run also `fetch_incidents` method.

Reply:
[adi recommendations](https://github.com/demisto/content/pull/23592/commits/4e5a2f1e787a6dbddfd39f955ab80c0564c8bca0)

Reply:
looks good
just change to test the integration always:

client.list_external_service_request([])

if params.get('isFetch'):  # Tests fetch incident:
            severity = params.get('severity')
            max_fetch = int(params.get('max_fetch', 10))
            fetch_incidents(
                client=client,
                max_fetch=max_fetch,
                last_run={},
                first_fetch_time=first_fetch_time,
                severity=severity
            )

Reply:
[another update to test_module](https://github.com/demisto/content/pull/23592/commits/4c527399388cc61a1b317634421bcac3bd94bba6)